web/src/beta/lib/core/StoryPanel/Page/index.tsx

```diff
@@ -47,82 +71,137 @@ const StoryPage: React.FC<Props> = ({
 }) => {
   const t = useT();
   const propertyItems = useMemo(() => page?.property.items, [page?.property]);
 
   const storyBlocks = useMemo(() => page?.blocks, [page?.blocks]);
 
-  const { openBlocksIndex, titleId, titleProperty, handleBlockOpen } = useHooks({
+  const {
+    openBlocksIndex,
+    titleId,
+    titleProperty,
+    isHovered,
+    handleOnMouseOver,
+    handleOnMouseOut,
+    handleBlockOpen,
+  } = useHooks({
     pageId: page?.id,
     propertyItems,
   });
 
+  const [items, setItems] = useState(storyBlocks ? storyBlocks : []);
+  const listRef = useRef<null | HTMLDivElement>(null);
+
+  useEffect(() => {
+    storyBlocks && setItems(storyBlocks);
+  }, [storyBlocks]);
+
+  const { updatePosition } = useScroll(listRef);
+
+  const dragDropManager = useDragDropManager();
+  const monitor = dragDropManager.getMonitor();
+
+  useEffect(() => {
+    const unsubscribe = monitor.subscribeToOffsetChange(() => {
+      const offset = monitor.getSourceClientOffset()?.y as number;
+      updatePosition({ position: offset, isScrollAllowed: true });
+    });
+    return unsubscribe;
+  }, [monitor, updatePosition]);
+
   return (
-    <SelectableArea
-      title={page?.title ?? t("Page")}
-      position="left-bottom"
-      icon="storyPage"
-      noBorder
-      isSelected={selectedPageId === page?.id}
-      propertyId={page?.property?.id}
-      propertyItems={propertyItems}
-      showSettings={showPageSettings}
-      isEditable={isEditable}
-      onClick={() => onPageSelect?.(page?.id)}
-      onClickAway={onPageSelect}
-      onSettingsToggle={onPageSettingsToggle}>
-      <Wrapper id={page?.id}>
-        {titleProperty && (
-          <StoryBlock
-            block={{
-              id: titleId,
-              pluginId: "reearth",
-              extensionId: "titleStoryBlock",
-              title: titleProperty.title,
-              property: {
-                id: page?.property?.id ?? "",
-                items: [titleProperty],
-              },
-            }}
-            isEditable={isEditable}
-            isSelected={selectedStoryBlockId === titleId}
-            onClick={() => onBlockSelect?.(titleId)}
-            onClickAway={onBlockSelect}
-            onChange={onPropertyUpdate}
-          />
-        )}
-        {isEditable && (
-          <BlockAddBar
-            openBlocks={openBlocksIndex === -1}
-            installableStoryBlocks={installableStoryBlocks}
-            onBlockOpen={() => handleBlockOpen(-1)}
-            onBlockAdd={onBlockCreate?.(0)}
-          />
-        )}
-        {storyBlocks?.map((b, idx) => (
-          <Fragment key={idx}>
-            <StoryBlock
-              block={b}
-              isSelected={selectedStoryBlockId === b.id}
-              isEditable={isEditable}
-              onClick={() => onBlockSelect?.(b.id)}
-              onClickAway={onBlockSelect}
-              onChange={onPropertyUpdate}
-              onRemove={onBlockDelete}
-            />
+    <DndProvider>
```

Comment:
DndProvider is already wrapping the Editor and Published components, so you shouldn't need to wrap here.